File: Arrays & Hashing/1. Two Sum/main.cpp
#include <unordered_map>
#include <vector>

class Solution {
public:
  std::vector<int> twoSum(std::vector<int> &nums, int target) {
    std::unordered_map<int, int> map;
    for (int i = 0; i < nums.size(); i++) {
      int targetNum = target - nums[i];
      if (map.count(targetNum)) {
        return {map[targetNum], i};
      }
      map[nums[i]] = i;
    }
    return {};
  }
};

File: Arrays & Hashing/1. Two Sum/main.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        dict = {}

        for i, num in enumerate(nums):
            if (target - num in dict):
                return [dict[target - num], i]
            dict[num] = i
        return [-1, -1]
